src/main/java/codeu/controller/RoutingEngine.java

```diff
@@ -0,0 +1,23 @@
+package codeu.controller;
+
+import codeu.model.data.BotRequest;
+
+/** This is the Routing Engine that will receive a BotRequest and pass it to the 
+ * appropriate BotHandler class*/
+public class RoutingEngine {
+
```

Comment:
Since it looks like you were planning each case to return the string, I think you might actually want to define something like

```java
BotHandlerInterface hander;
switch (request.getCommand()) {
  case "foo":
    hander = FooHander();
  default:
    throw Error
}
hander.handle(botRequest);
```

Reply:
This mean you won't repeat the `.handle(...)` call for every handler, since they all need to satisfy the interface anyways